### PR TITLE
feat: Use Semconv Naming For ActionPack

### LINKED ISSUE #961: Use semconv span naming in ActionPack
`action_pack` currently overrides the `server` span name to a value that does not align with semantic conventions. Per the specification[^1]
 
> HTTP spans MUST follow the overall [guidelines for span names](https://github.com/open-telemetry/opentelemetry-specification/tree/v1.31.0/specification/trace/api.md#span).
> 
> HTTP server span names SHOULD be {method} {http.route} if there is a (low-cardinality) http.route available (see below for the exact definition of the [{method}](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#method-placeholder) placeholder).
> 
> If there is no (low-cardinality) http.route available, HTTP server span names SHOULD be [{method}](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#method-placeholder).
> 
> HTTP client spans have no http.route attribute since client-side instrumentation is not generally aware of the “route”, and therefore HTTP client span names SHOULD be [{method}](https://opentelemetry.io/docs/specs/semconv/http/http-spans/#method-placeholder).
> 
> The {method} MUST be {http.request.method} if the method represents the original method known to the instrumentation. In other cases (when {http.request.method} is set to _OTHER), {method} MUST be HTTP.
> 
> Instrumentation MUST NOT default to using URI path as span name, but MAY provide hooks to allow custom logic to override the default span name.

In this case the span name is set to RubyDoc naming style, with the class name of the controller followed by an `#` then the action:

https://github.com/open-telemetry/opentelemetry-ruby-contrib/blob/10e766e13e7734f284807af4099e0858ca85cf4f/instrumentation/action_pack/lib/opentelemetry/instrumentation/action_pack/handlers/action_controller.rb#L29

I propose the span name be set to either of the following values:

- `>= 7.1` [`ActionDispatch::Request#route_uri_pattern`](https://api.rubyonrails.org/classes/ActionDispatch/Request.html#method-i-route_uri_pattern) 
- `< 7.1` `#{method} /#{underscore(controller)}/#{action}`

In addition to that, we should restore the `http.route` attribute. 

[^1]: https://opentelemetry.io/docs/specs/semconv/http/http-spans/#name

### PR DESCRIPTION
**Pull request by arielvalentin**
Aligns the span name closer to the specification instead of using Ruby class/method naming syntax

Fixes https://github.com/open-telemetry/opentelemetry-ruby-contrib/issues/961

### CODE REVIEW

File: instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb [hunk 1]
```diff
@@ -65,7 +64,7 @@
 
     _(last_response.body).must_equal 'created new item'
     _(last_response.ok?).must_equal true
-    _(span.name).must_equal 'ExampleController#new_item'
+    # _(span.name).must_equal 'GET example/new_item'
```
Comment (MikeGoldsmith) [inline instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb:67]: Should these be updated or removed?

File: instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb [hunk 2]
```diff
@@ -139,13 +139,79 @@
...
+  end
+
   describe 'when the application has exceptions_app configured' do
     let(:rails_app) { AppConfig.initialize_app(use_exceptions_app: true) }
 
     it 'does not overwrite the span name from the controller that raised' do
       get 'internal_server_error'
 
-      _(span.name).must_equal 'ExampleController#internal_server_error'
+      # _(span.name).must_equal 'GET example/show'
```
Comment (MikeGoldsmith) [inline instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb:214]: Should these be updated or removed?

**Review by MikeGoldsmith — APPROVED** [review]
Looks good to me. Added a couple of queries for commented out assertions.

File: instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb (hunk 1 above)
Comment (arielvalentin) [reply instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb:67]: Oh my. That definitely was a mistake on my part.

Thank you for pointing this out!

File: instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb (hunk 2 above)
Comment (arielvalentin) [reply instrumentation/action_pack/test/opentelemetry/instrumentation/action_pack/handlers/action_controller_test.rb:214]: 😳

**Review by xuan-cao-swi — APPROVED** [review]
👍

**Pull request merged**